stdlib/SparseArrays/src/linalg.jl

```diff
@@ -203,6 +203,28 @@ function spmatmul(A::SparseMatrixCSC{Tv,Ti}, B::SparseMatrixCSC{Tv,Ti};
     return C
 end
 
+# Frobenius inner product: trace(A†B)
+function vecdot(A::SparseMatrixCSC{T1,S1},B::SparseMatrixCSC{T2,S2}) where {T1,T2,S1,S2}
+    m, n = size(A)
+    size(B) == (m,n) || throw(DimensionMismatch("matrices must have the same dimensions"))
+    r = zero(promote_type(T1,T2))
+    @inbounds for j = 1:n
+        for i1 = A.colptr[j]:A.colptr[j+1]-1
+            ra = A.rowval[i1]
+            for i2 = B.colptr[j]:B.colptr[j+1]-1
+                rb = B.rowval[i2]
```

Comment:
I realise it's a bit pedantic, but maybe keep the indices consistent? i.e. `ia` instead of `i1` and `ib` instead of `i2`.